pandas/core/internals/blocks.py

```diff
@@ -1341,6 +1341,8 @@ def func(cond, values, other):
                 return values
 
             values, other = self._try_coerce_args(values, other)
+            if isinstance(values, ExtensionArray):
+                values = values.get_values().reshape(-1, 1)  # GH 16983
```

Comment:
Just FYI, we want to avoid `get_values` since that converts the ExtensionArray to an ndarray.